node/src/service/service_parachain.rs

```diff
@@ -65,7 +65,7 @@ pub fn new_partial(
         config.transaction_pool.clone(),
         config.role.is_authority().into(),
         config.prometheus_registry(),
-        task_manager.spawn_handle(),
+        task_manager.spawn_essential_handle(),
```

Comment:
Does this prevent a client from further operating a tx pool when an error occurs, i.e. the tx pool is shut down (and therefore the client stops operating, at least this service)?

Reply:
I don't know the difference between an ordinary and an essential task. This code was simply copied and pasted from the `0.9.6` cumulus  branch.
https://core.tetcoin.org/rustdocs/v2.0.1/sc_service/struct.TaskManager.html does not state the difference and https://github.com/paritytech/cumulus/pull/496 does not tell why `spawn_handle` was replaced.
A spawned task is a separated unit of computation (`T: Send`) that lives with its own context and I don't know a failed task will prevent the client from further operating a tx pool. If you see https://github.com/paritytech/substrate/blob/4d9f03d3d6d0ceec0708b1a23978feee12f5b0bb/client/service/src/task_manager/mod.rs#L149-L157, then it might be possible that `spawn_essential_handle` will improve the robustness of the node.